include/fschuetz04/simcpp20/event.hpp

```diff
@@ -409,7 +409,8 @@ template <typename Time = double> class event {
     /// Destructor.
     virtual ~data() {
       for (auto &handle : handles_) {
-        handle.destroy();
+        if (handle)
+          handle.destroy();
```

Comment:
I would not add this check here, as this might hide errors in the future. The other change is sufficient.